pkg/distributor/distributor_test.go

```diff
@@ -292,6 +293,22 @@ func TestDistributor_Push(t *testing.T) {
 				cortex_distributor_latest_seen_sample_timestamp_seconds{user="user"} 123456789.009
 			`,
 		},
+		"A push to 3 happy ingesters using batch worker gouroutines should succeed": {
```

Comment:
WDYT if we add another test in which we show, similarly to what is done in `dskit`, that setting `cfg.IngesterPushWorkerGoroutines` to a positive value, actually uses that number of workers?

Reply:
I don't like the idea of counting goroutines here, with all the services and stuff going on, that can easily break. However, I've added another test below that wraps the workers with a function that counts the calls and checks the counter. This also served me to see that service wasn't properly created, this test wasn't calling `configure`, and workers weren't actually used :facepalm:

Reply:
Thank you for adding this test.